src/transports/TransportRequestManager.ts

```diff
@@ -122,17 +124,19 @@ export class TransportRequestManager {
   private resolveRes(data: IJSONRPCNotificationResponse | IJSONRPCResponse, emitError: boolean): TransportResponse {
     const { id, error } = data;
 
-    const status = this.pendingRequest[id as string];
-    if (status) {
-      delete this.pendingRequest[id as string];
-      this.processResult(data, status);
-      this.transportEventChannel.emit("response", data as IJSONRPCResponse);
-      return;
-    }
     if (id === undefined && error === undefined) {
       this.transportEventChannel.emit("notification", data as IJSONRPCNotificationResponse);
       return;
     }
+    const status = this.pendingRequest[id as string];
+    if (status) {
+      delete this.pendingRequest[id as string];
+      if (error === undefined) {
```

Comment:
This is the key change. We may have an error response, which per the spec should result in an error response. The prior code block only checked to see if we were expecting the response with a particular ID, and early-returned. The `processResult` method would still reject the promise, however it would not return an error object.